GrayBMP.cs

```diff
@@ -74,6 +74,17 @@ class GrayBMP {
       Dirty (x1, y1); Dirty (x2, y2);
    }
 
+   public void DrawHorizontalLine (int x1, int x2, int y, int gray) {
+      Begin ();
+      Check (x1, y); Check (x2, y); Dirty (x1, y, x2, y);
+      if (x1 > x2) (x1, x2) = (x2, x1);
+      unsafe {
+         byte* ptr = (byte*)(Buffer + y * mStride + x1);
+         System.Runtime.CompilerServices.Unsafe.InitBlock (ref *ptr, (byte)gray, (uint)(x2 - x1));

```

Comment:
Typically, x2 - x1 is a small number of bytes, and a for loop might actually be faster than InitBlock for such small runs. Also, you are drawing one pixel too less (if I do a `DrawHorizontalLine (10, 5, 20, 5, 255)` I am expecting **11** pixels to get set).